### PR TITLE
Add quality type to ufp files

### PR DESCRIPTION
**Pull request by casperlamboo**
Cura-9773

### CODE REVIEW

File: cura/Settings/MachineManager.py [hunk 1]
```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
...
+    def activeQualityDisplayNameStringParts(self) -> [str]:
+        result_map = self.activeQualityDisplayNameMap
+        string_parts = list()
+
+        if result_map["custom_profile"] is not None:
+            string_parts.append(result_map["custom_profile"])
 
-        display_name = global_stack.quality.getName()
+        if result_map["intent_category"] is not "default":
+            string_parts.append(f"""{result_map["intent_name"]} - {result_map["profile"]}""")
```
Comment (Joeydelarago) [inline cura/Settings/MachineManager.py:1643]:
```suggestion
            string_parts.append(f"{result_map["intent_name"]} - {result_map["profile"]}")
```

File: cura/Settings/MachineManager.py [hunk 2]
```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
...
-    @pyqtProperty("QVariantMap", notify = activeQualityDisplayNameChanged)
-    def activeQualityDisplayNameMap(self) -> Dict[str, str]:
-        global_stack = self._application.getGlobalContainerStack()
-        if global_stack is None:
-            return {"main": "",
-                    "suffix": ""}
+    @pyqtProperty("QList<QString>", notify = activeQualityDisplayNameChanged)
+    def activeQualityDisplayNameStringParts(self) -> [str]:
+        result_map = self.activeQualityDisplayNameMap
+        string_parts = list()
```
Comment (Joeydelarago) [inline cura/Settings/MachineManager.py:1637]:
```suggestion
        string_parts = []
```

Pythonic & faster to initialise like this

File: cura/Settings/MachineManager.py [hunk 3]
```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
...
 
-        intent_category = self.activeIntentCategory
-        if intent_category != "default":
-            intent_display_name = IntentCategoryModel.translation(intent_category,
-                                                                  "name",
-                                                                  intent_category.title())
-            display_name = "{intent_name} - {the_rest}".format(intent_name = intent_display_name,
-                                                               the_rest = display_name)
+        if result_map["layer_height"]:
+            string_parts.append(f"""{result_map["layer_height"]}mm""")
```
Comment (Joeydelarago) [inline cura/Settings/MachineManager.py:1648]:
```suggestion
            string_parts.append(f"{result_map["layer_height"]}mm")
```

File: cura/Settings/MachineManager.py [hunk 4]
```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
...
     #          - "my_profile - Fine" (only based on a default quality, no intent involved)
     #          - "my_profile - Engineering - Fine" (based on an intent)
-    @pyqtProperty("QVariantMap", notify = activeQualityDisplayNameChanged)
-    def activeQualityDisplayNameMap(self) -> Dict[str, str]:
-        global_stack = self._application.getGlobalContainerStack()
-        if global_stack is None:
-            return {"main": "",
-                    "suffix": ""}
+    @pyqtProperty("QList<QString>", notify = activeQualityDisplayNameChanged)
+    def activeQualityDisplayNameStringParts(self) -> [str]:
```
Comment (Joeydelarago) [inline cura/Settings/MachineManager.py:1635]:
```suggestion
    def activeQualityDisplayNameStringParts(self) -> List[str]:
```

You'll need the import too

**Review by Joeydelarago — CHANGES_REQUESTED** [review]

File: cura/Settings/MachineManager.py (hunk 3 above)
Comment (casperlamboo) [reply cura/Settings/MachineManager.py:1648]: I tried this, but this gave me linting errors in pycharm

File: cura/Settings/MachineManager.py (hunk 4 above)
Comment (casperlamboo) [reply cura/Settings/MachineManager.py:1635]: List is already imported 👍

**Review by Joeydelarago — APPROVED** [review]

**Pull request merged**